### PR TITLE
Feat5/update music player component(play/pause, next, prev)

### PR DESCRIPTION
**Pull request by jiayuu-u**
### Descriptions
Make the play/pause, forward and previous buttons functional.

- Link to UI design (if possible):

### Provide screenshots or videos that the function is working.

https://user-images.githubusercontent.com/68804519/234173421-f9305a2c-6678-46a0-9049-5117a73a74ca.mov

### Provide screenshots of no warning, or error in the console.

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

File: src/components/AudioControls.tsx [hunk 1]
```diff
@@ -67,17 +67,28 @@ const AudioControls = ({
...
         </div>
 
-        {isPlaying ? (
-          <button type="button" className="items-center justify-center" onClick={onPlayPauseClick}>
+        {playingQueue && isPlaying ? (
+          <button
+            type="button"
+            className="items-center justify-center"
+            onClick={onPlayPauseClick}
+            role="playButton"
```
Comment (Cameron933) [inline src/components/AudioControls.tsx:81]: when you doing the test, findAllByRole("button")[2] is another option so you do not need to add testing-id or role to the component which you are testing.

File: src/layouts/MusicPlayer.tsx [hunk 2]
```diff
@@ -44,6 +45,25 @@ const MusicPlayer = () => {
     if (loopMode === "all") setLoopMode("none");
   };
 
+  const onPrevClick = () => {
+    if (playingIndex === 0) {
```
Comment (Cameron933) [inline src/layouts/MusicPlayer.tsx:49]: change the if-else to Ternary

File: src/layouts/MusicPlayer.tsx [hunk 3]
```diff
@@ -44,6 +45,25 @@ const MusicPlayer = () => {
...
+      let lastTrackIndex = musicData.length - 1;
+      setPlayingIndex(lastTrackIndex);
+    } else {
+      setPlayingIndex((prev: number) => prev - 1);
+    }
+  };
+
+  const onNextClick = () => {
+    if (playingIndex >= musicData.length - 1) {
```
Comment (Cameron933) [inline src/layouts/MusicPlayer.tsx:58]: same

File: src/layouts/MusicPlayer.tsx [hunk 4]
```diff
@@ -59,6 +79,16 @@ const MusicPlayer = () => {
...
   }, [isPlaying, audioUrl]);
 
+  useEffect(() => {
+    {
+      isPlaying ? audioRef.current?.play() : audioRef.current?.pause();
+    }
+  }, [isPlaying, audioRef]);
+
+  const onPlayPauseClick = () => {
+    setIsPlaying((prev) => !prev);
```
Comment (Cameron933) [inline src/layouts/MusicPlayer.tsx:89]: Why you do not just make the isPlaying a useState()?

Comment (Cameron933) [reply src/layouts/MusicPlayer.tsx:89]: plus, the prev is confusing here

**Review by Cameron933 — APPROVED** [review]

**Comment by Double-Twelve** [comment]
In the `onPrevClick`, `onNextClick`, `useEffect({},[isPlaying, audioRef])` three methods
1. why wrap them with another curly braces?
2. need to add a if statement, if `playingIndex !== -1`, and then run the following code

File: src/layouts/MusicPlayer.tsx [hunk 5]
```diff
@@ -95,7 +122,14 @@ const MusicPlayer = () => {
             <p className="text-sm"> - </p>
             <p className="text-xs">Jaxson Westervelt</p>
           </div>
-          <AudioControls loopMode={loopMode} toggleLoopMode={toggleLoopMode} />
+          <AudioControls
+            loopMode={loopMode}
+            toggleLoopMode={toggleLoopMode}
+            isPlaying={isPlaying}
```
Comment (Double-Twelve) [inline src/layouts/MusicPlayer.tsx:128]: `isPlaying` can be got from the context, don't need to pass

**Review by Double-Twelve — APPROVED** [review]
I have reviewed the changes and everything looks good to me. Thanks for your contribution.

**Pull request merged**